jerry-ext/module/module.c

```diff
@@ -202,29 +214,51 @@ jerryx_module_native_resolver (const jerry_char_t *name, /**< name of the module
  *   - an error indicating that something went wrong during the attempt to load the module.
  */
 jerry_value_t
-jerryx_module_resolve (const jerry_char_t *name, /**< name of the module to load */
-                       const jerryx_module_resolver_t *resolvers_p, /**< list of resolvers */
+jerryx_module_resolve (const jerry_value_t name, /**< name of the module to load */
+                       const jerryx_module_resolver_t **resolvers_p, /**< list of resolvers */
                        size_t resolver_count) /**< number of resolvers in @p resolvers */
 {
   size_t index;
+  size_t canonical_names_used = 0;
   jerry_value_t ret;
-  jerry_value_t instances = *(jerry_value_t *) jerry_get_context_data (&jerryx_module_manager);
-  jerry_value_t module_name = jerry_create_string_from_utf8 (name);
+  const jerryx_module_resolver_t *resolver_p;
+  jerry_value_t instances;
+  jerry_value_t canonical_names[resolver_count];
 
-  /* Return the cached instance if present. */
-  if (jerryx_module_check_cache (instances, module_name, &ret))
+  if (!jerry_value_is_string (name))
   {
+    ret = jerryx_module_create_error (JERRY_ERROR_COMMON, module_name_not_string, name);
     goto done;
   }
 
+  instances = *(jerry_value_t *) jerry_get_context_data (&jerryx_module_manager);
+
+  /**
+   * Establish the canonical name for the requested module. Each resolver presents its own canonical name. If one of
+   * the canonical names matches a cached module, it is returned as the result.
+   */
+  for (index = 0; index < resolver_count; index++)
+  {
+    resolver_p = resolvers_p[index];
+    canonical_names[index] = ((resolver_p && resolver_p->get_canonical_name != NULL) ?
+      resolver_p->get_canonical_name (name):
+      jerry_acquire_value (name));
```

Comment:
This is not how we align ?:

Reply:
It's pretty tough not breaking 120 columns with this line. I'm still trying to figure out how to align it.

Reply:
I renamed `resolver_p` to `res_p` then it fits.

Reply:
I would move the condition into a bool value rather than shoretning the name.